rackcity/models/asset.py

```diff
@@ -250,7 +289,18 @@ class AssetCP(AbstractAsset):
         ITModel, on_delete=models.SET_NULL, null=True, verbose_name="related model",
     )
     rack = models.ForeignKey(
-        Rack, on_delete=models.SET_NULL, null=True, verbose_name="related rack",
+        Rack,
+        on_delete=models.SET_NULL,
+        verbose_name="related rack",
+        null=True,
+        blank=True,
+    )
+    chassis = models.ForeignKey(
+        "self",
+        on_delete=models.PROTECT,
+        verbose_name="blade chassis",
+        null=True,
+        blank=True,
```

Comment:
I think `on_delete` should be SET_NULL?

Reply:
This reference is to another AssetCP, so I think we can enforce within the change plan that the chassis (in the CP) won't be deleted if it has blades (in the CP) in it

Reply:
oof ok more of a side note then: when detecting conflicts on change plan, every blade is going to need to check if it's chassis has any conflicts (specifically if the live chassis has been deleted)